### PR TITLE
Localization: Localized Gacha Machine Names

### PR DESCRIPTION
**Pull request by josericardo-fo**
## What are the changes?

Localized the names of the gacha machines.
## Why am I doing these changes?

They weren't localized yet.
## What did change?

Localization and the names of the gachas.
### Screenshots/Videos

# DE
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# ES
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# FR
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# KO
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# PT_BR
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# ZH_CN
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

# ZH_TW
## Move UP!
image

## Legendary UP!
image

## Shiny UP!
image

## How to test the changes?

See if the Gacha Machine name fits in the Gacha Menu.
## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/zh_TW/egg.ts [hunk 1]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "{{pokemonName}} 從蛋中孵化了！",
   "eggMoveUnlock": "蛋招式已解鎖： {{moveName}}",
   "rareEggMoveUnlock": "稀有蛋招式已解鎖： {{moveName}}",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
+  "legendaryUPGacha": "UP!",
```
Comment (40chyan) [inline src/locales/zh_TW/egg.ts:27]:
```suggestion
  "moveUPGacha": "蛋招式UP!",
  "shinyUPGacha": "闪光概率UP!,
  "legendaryUPGacha": "UP!",
```
zh_cn

**Comment by flx-sta** [comment]
Well that translation broke the build

File: src/locales/zh_TW/egg.ts [hunk 2]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "{{pokemonName}} 從蛋中孵化了！",
   "eggMoveUnlock": "蛋招式已解鎖： {{moveName}}",
   "rareEggMoveUnlock": "稀有蛋招式已解鎖： {{moveName}}",
+  "moveUPGacha": "蛋招式UP!",
+  "shinyUPGacha": "闪光概率UP!,
```
Comment (EnochG1) [inline src/locales/zh_TW/egg.ts:26]:
```suggestion
  "shinyUPGacha": "闪光概率UP!",
```

**Review by EnochG1 — COMMENTED** [review]
Let's fix it at first.

File: src/locales/de/egg.ts [hunk 3]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "Ein {{pokemonName}} ist aus dem Ei geschlüpft!",
   "eggMoveUnlock": "Ei-Attacke freigeschaltet: {{moveName}}",
   "rareEggMoveUnlock": "Seltene Ei-Attacke freigeschaltet: {{moveName}}",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
+  "legendaryUPGacha": "UP!",
```
Comment (CodeTappert) [inline src/locales/de/egg.ts:27]:
```suggestion
  "moveUPGacha": "Attacken-UP!",
  "shinyUPGacha": "Schillernd-UP!",
  "legendaryUPGacha": "UP!",
```

File: src/locales/es/egg.ts [hunk 4]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "¡Ha salido un {{pokemonName}} del Huevo!",
   "eggMoveUnlock": "Mov. Huevo desbloqueado:\n{{moveName}}",
   "rareEggMoveUnlock": "Mov. Huevo Raro desbloqueado:\n{{moveName}}",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
+  "legendaryUPGacha": "UP!",
```
Comment (InnocentGameDev) [inline src/locales/es/egg.ts:27]:
```suggestion
  "moveUPGacha": "¡Más Mov. Huevo Raro!",
  "shinyUPGacha": "¡Más Shiny!",
  "legendaryUPGacha": "más probable!",
```

File: src/locales/fr/egg.ts [hunk 5]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "{{pokemonName}} sort de l’Œuf !",
   "eggMoveUnlock": "Capacité Œuf débloquée :\n{{moveName}}",
   "rareEggMoveUnlock": "Capacité Œuf Rare débloquée :\n{{moveName}}",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
+  "legendaryUPGacha": "UP!",
```
Comment (Adri1) [inline src/locales/fr/egg.ts:27]:
```suggestion
  "moveUPGacha": "Capa + !",
  "shinyUPGacha": "Chroma + !",
  "legendaryUPGacha": "+ !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/ko/egg.ts [hunk 6]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "알이 부화해서\n{{pokemonName}}[[가]] 태어났다!",
   "eggMoveUnlock": "알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다!",
   "rareEggMoveUnlock": "레어 알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다!",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
```
Comment (sodaMelon) [inline src/locales/ko/egg.ts:26]:
```suggestion
  "moveUPGacha": "기술 UP!",
  "shinyUPGacha": "특별색 UP!",
```

**Review by sodaMelon — COMMENTED** [review]
here is the korean translation. ( it was decided in korean discord )

File: src/locales/es/egg.ts (hunk 4 above)
Comment (josericardo-fo) [reply src/locales/es/egg.ts:27]: The text needs to be smaller

File: src/locales/zh_TW/egg.ts [hunk 7]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "{{pokemonName}} 從蛋中孵化了！",
   "eggMoveUnlock": "蛋招式已解鎖： {{moveName}}",
   "rareEggMoveUnlock": "稀有蛋招式已解鎖： {{moveName}}",
+  "moveUPGacha": "蛋招式UP!",
+  "shinyUPGacha": "闪光概率UP!",
```
Comment (40chyan) [inline src/locales/zh_TW/egg.ts:26]:
```suggestion
  "shinyUPGacha": "概率UP!",
```
fixed

Comment (40chyan) [inline src/locales/zh_TW/egg.ts:26]:
```suggestion
  "shinyUPGacha": "闪光UP!",
```

File: src/locales/zh_CN/egg.ts [hunk 8]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "{{pokemonName}} 从蛋中孵化了！",
   "eggMoveUnlock": "蛋招式已解锁: {{moveName}}",
   "rareEggMoveUnlock": "稀有蛋招式已解锁: {{moveName}}",
+  "moveUPGacha": "Move UP!",
+  "shinyUPGacha": "Shiny UP!",
+  "legendaryUPGacha": "UP!",
```
Comment (40chyan) [inline src/locales/zh_CN/egg.ts:27]:
```suggestion
  "moveUPGacha": "蛋招式UP!",
  "shinyUPGacha": "闪光UP!",
  "legendaryUPGacha": "UP!",
```

**Review by Adri1 — APPROVED** [review]

File: src/locales/es/egg.ts (hunk 4 above)
Comment (flx-sta) [reply src/locales/es/egg.ts:27]: Shorter*

**Review by EnochG1 — APPROVED** [review]

File: src/locales/es/egg.ts (hunk 4 above)
Comment (InnocentGameDev) [reply src/locales/es/egg.ts:27]: How about?

```suggestion
  "moveUPGacha": "Mov. H. +!",
  "shinyUPGacha": "Shiny +!",
  "legendaryUPGacha": "+!",
```

Comment (josericardo-fo) [reply src/locales/es/egg.ts:27]: Works just fine

Comment (flx-sta) [reply src/locales/es/egg.ts]: what the heck? legendary is just `+!`?

Comment (Adri1) [reply src/locales/es/egg.ts]: > what the heck? legendary is just `+!`?

The legendary it the icon of the Pokémon and then the text, so what you see here is only the text part

File: src/ui/egg-gacha-ui-handler.ts [hunk 9]
```diff
@@ -100,7 +100,7 @@ export default class EggGachaUiHandler extends MessageUiHandler {
       let gachaTextStyle = TextStyle.WINDOW_ALT;
       let gachaX = 4;
       let gachaY = 0;
-      if (currentLanguage === "de" || "pt-br") {
+      if (currentLanguage === "de" || currentLanguage === "fr" || currentLanguage === "pt-BR") {
```
Comment (flx-sta) [inline src/ui/egg-gacha-ui-handler.ts]:
```suggestion
      if (["de", "fr", "pt-BR"].includes(currentLanguage)) {
```

File: src/ui/egg-gacha-ui-handler.ts [hunk 10]
```diff
@@ -119,19 +119,21 @@ export default class EggGachaUiHandler extends MessageUiHandler {
         gachaInfoContainer.add(pokemonIcon);
         break;
       case GachaType.MOVE:
-        if (currentLanguage === "de") {
+        if (currentLanguage === "de" || currentLanguage === "fr") {
```
Comment (flx-sta) [inline src/ui/egg-gacha-ui-handler.ts]:
```suggestion
        if (["de", "fr"].includes(currentLanguage) {
```

File: src/ui/egg-gacha-ui-handler.ts [hunk 11]
```diff
@@ -119,19 +119,21 @@ export default class EggGachaUiHandler extends MessageUiHandler {
...
           gachaUpLabel.setY(0);
         }
+
         gachaUpLabel.setText(i18next.t("egg:moveUPGacha"));
         gachaUpLabel.setX(0);
         gachaUpLabel.setOrigin(0.5, 0);
         break;
       case GachaType.SHINY:
-        if (currentLanguage === "de") {
+        if (currentLanguage === "de" || currentLanguage === "fr") {
```
Comment (flx-sta) [inline src/ui/egg-gacha-ui-handler.ts:132]:
```suggestion
        if (["de", "fr"].includes(currentLanguage)) {
```

**Review by flx-sta — CHANGES_REQUESTED** [review]

**Review by edaymall — APPROVED** [review]

**Review by edaymall — APPROVED** [review]

**Review by sodaMelon — APPROVED** [review]

**Comment by sodaMelon** [comment]
edaymall is my friend's account. don't mind. (just click mistake)
there's no problem in Korean Localization

File: src/ui/egg-gacha-ui-handler.ts (hunk 11 above)
Comment (josericardo-fo) [reply src/ui/egg-gacha-ui-handler.ts]: fixed

File: src/locales/de/egg.ts [hunk 12]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "Ein {{pokemonName}} ist aus dem Ei geschlüpft!",
   "eggMoveUnlock": "Ei-Attacke freigeschaltet: {{moveName}}",
   "rareEggMoveUnlock": "Seltene Ei-Attacke freigeschaltet: {{moveName}}",
+  "moveUPGacha": "Mehr\nEi-Attacken!",
+  "shinyUPGacha": "Mehr\nSchillernde!",
+  "legendaryUPGacha": "UP!",
```
Comment (CodeTappert) [inline src/locales/de/egg.ts:27]:
```suggestion
  "legendaryUPGacha": "erscheint\nöfter!",
```

Comment (josericardo-fo) [reply src/locales/de/egg.ts]: committed

File: src/locales/es/egg.ts [hunk 13]
```diff
@@ -22,4 +22,7 @@ export const egg: SimpleTranslationEntries = {
   "hatchFromTheEgg": "¡Ha salido un {{pokemonName}} del Huevo!",
   "eggMoveUnlock": "Mov. Huevo desbloqueado:\n{{moveName}}",
   "rareEggMoveUnlock": "Mov. Huevo Raro desbloqueado:\n{{moveName}}",
+  "moveUPGacha": "¡Más Mov.\nHuevo Raro!",
+  "shinyUPGacha": "¡Más Shiny!",
+  "legendaryUPGacha": "+!",
```
Comment (InnocentGameDev) [inline src/locales/es/egg.ts:27]:
```suggestion
  "legendaryUPGacha": ", ¡mayor prob.!",
```

**Review by InnocentGameDev — APPROVED** [review]
The legendary one is the one that seems a bit off, otherwise it's great. Thanks for your work!.

**Review by Tempo-anon — APPROVED** [review]

**Comment by josericardo-fo** [comment]
I didn't mean to revert it, just ignore this

**Pull request merged**